l1-contracts/contracts/zksync/facets/Executor.sol

```diff
@@ -472,4 +497,92 @@ contract ExecutorFacet is Base, IExecutor {
+    }
+
+    /// Verifies that the blobs contain the correct data by calling the point evaluation precompile. For the precompile we need:
+    /// versioned hash || opening point || opening value || commitment || proof
+    /// the _pubdataCommitments will contain the last 4 values, the versioned hash is pulled from the BLOBHASH opcode
+    /// pubdataCommitments is a list of: opening point (16 bytes) || claimed value (32 bytes) || commitment (48 bytes) || proof (48 bytes)) = 144 bytes
+    function _verifyBlobInformation(
+        bytes calldata _pubdataCommitments,
+        bytes32[2] memory _blobHashes
+    ) internal view returns (bytes32[] memory blobCommitments) {
+        uint256 versionedHashIndex = 0;
+
+        // TODO: This should be dynamic instead of being hardcoded to 2 blobs
+        require(_pubdataCommitments.length > 0, "pl");
+        require(
+            _pubdataCommitments.length <= PUBDATA_COMMITMENT_SIZE * 2 &&
+                _pubdataCommitments.length % PUBDATA_COMMITMENT_SIZE == 0,
+            "bs"
+        );
+        blobCommitments = new bytes32[](2);
+        bytes32 versionedHash;
+
+        for (uint256 i = 0; i < _pubdataCommitments.length; i += PUBDATA_COMMITMENT_SIZE) {
+            versionedHash = _getBlobVersionedHash(versionedHashIndex);
+
+            require(versionedHash != bytes32(0), "vh");
+
+            // First 16 bytes is the opening point. While we get the point as 16 bytes, the point evaluation precompile
+            // requires it to be 32 bytes. The blob commitment must use the opening point as 16 bytes though.
+            bytes32 openingPoint = bytes32(uint256(uint128(bytes16(_pubdataCommitments[i: i + PUBDATA_COMMITMENT_CLAIMED_VALUE_OFFSET]))));
+
+            _pointEvaluationPrecompile(
+                versionedHash,
+                openingPoint,
+                _pubdataCommitments[i + PUBDATA_COMMITMENT_CLAIMED_VALUE_OFFSET:i + PUBDATA_COMMITMENT_SIZE]
+            );
+
+            // Take the hash of the versioned hash || opening point || claimed value
+            blobCommitments[versionedHashIndex] = keccak256(
+                abi.encodePacked(
+                    versionedHash, 
+                    _pubdataCommitments[i:i + PUBDATA_COMMITMENT_COMMITMENT_OFFSET]
+                )
+            );
+            versionedHashIndex += 1;
+        }
+
+        // This check is required because we want to ensure that there arent any extra blobs trying to be published.
+        // Calling the BLOBHASH opcode with an index > # blobs - 1 yields bytes32(0)
+        versionedHash = _getBlobVersionedHash(versionedHashIndex);
+        require(versionedHash == bytes32(0), "lh");
+
+        // We do this check to verify that either the second blob was empty/not used
+        // or that we have both a linear hash and commitment for it.
+        require(
+            (_blobHashes[1] == bytes32(0) && blobCommitments[1] == bytes32(0)) ||
```

Comment:
Nit: 

This code will raise a question of why we don't check the same for `blobHash[0]`. The correct answer is that since we checked at the start of this function `_pubdataCommitments.length > 0,`, it is not possible that the first batch's commitment has not been processed.

While it is true, I would suggest for a more plain logic to have a for loop that would check it for each batch or add a comment that would explain that it is not possible that the first batch is empty

Reply:
ill add a comment so we dont add redundant code